packages/query-core/src/__tests__/queryClient.test-d.tsx

```diff
@@ -496,4 +496,24 @@ describe('fully typed usage', () => {
     queryClient.setQueryDefaults(queryKey, {} as any)
     queryClient.getMutationDefaults(mutationKey)
   })
+
+  it('shows type error when queryKey is a wrong type in invalidateQueries', () => {
+    const queryClient = new QueryClient()
+
+    queryClient.invalidateQueries()
+
+    queryClient.invalidateQueries({
+      queryKey: ['1'],
+    })
+
+    queryClient.invalidateQueries({
+      // @ts-expect-error
+      queryKey: '1',
+    })
+
+    queryClient.invalidateQueries({
+      // @ts-expect-error
+      queryKey: {},
+    })
+  })
```

Comment:
I’ve added this test to:
- https://github.com/TanStack/query/pull/8670

where I’m also re-writing the types for `invalidateQueries` and other methods.

the test passes there as well, so I’ll close this PR to avoid the conflicts.